Finish tag system

interrrp opened the pull request:
* Friend suggestions
* `/tag add` and `/tag remove`

Commit:
feat: add remaining tag system functionality

* Friend suggestions
* `/tag add` and `/tag remove`

sstollenwerk commented on bot/repositories/tags.py:
collections.Counter.most_common seems cleaner for the # Limit to top 10 users with most common tags
Not sure if it'd actually improve speed or memory.

interrrp replied:
Feel free to do it, I'm all for clean code but don't know how to implement your suggestion 😅

sstollenwerk committed:
refactor: get_friend_suggestions
basic tests added

sstollenwerk committed:
chore:allow assert statement in tests

sstollenwerk committed:
chore: obey INP001 in tests

sstollenwerk committed:
refactor: move old code from prod to tests

Commit:
fix: type errors

Commit:
refactor: move `TagType` to tag repository module

Beardrop2 committed:
fix: async test

The suggested friends function must be an async function so that the return the dict properly.
Also added pytest-asyncio and hypothesis so that pytest can run the tests.

Beardrop2 reviewed (approved):
Looks good.
Thanks for adding a test.

unseeyou committed:
Merge branch 'main' into finish-tag-system
